### PR TITLE
IGNITE-14039 Add warning to runtime, javadoc and documentation on WAL enable/disable.

### PR DESCRIPTION
**Pull request by alamar**
Also fixed WAL being incorrectly considered disabled after cache was destroyed and recreated.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/IgniteCluster.java [hunk 1]
```diff
@@ -527,6 +527,12 @@
      * when all dirty pages are prepared for checkpoint, but not necessarily flushed to disk.
      * <p>
      * WAL state can be changed only for persistent caches.
+     * <p>
+     * <b>NOTE:</b>
+     * Currently, this method should only be called on a stable topology when no nodes are leaving or joining cluster,
```
Comment (Mmuzaf) [inline modules/core/src/main/java/org/apache/ignite/IgniteCluster.java:532]: @alamar should we also add todo comment with the reference to the original issue?

File: modules/core/src/main/java/org/apache/ignite/IgniteCluster.java [hunk 2]
```diff
@@ -527,6 +527,12 @@
...
+     * and all baseline nodes are present.
+     * Cache may be stuck in inconsistent state due to violation of these conditions. It is advised to destroy
+     * such cache.
```
Comment (Mmuzaf) [inline modules/core/src/main/java/org/apache/ignite/IgniteCluster.java:535]:
```suggestion
     * such cache.
    TODO: IGNITE-13976 WAL disable/enable with node restarts results in mismatching state, data loss
```

File: modules/core/src/main/java/org/apache/ignite/IgniteCluster.java [hunk 3]
```diff
@@ -545,6 +551,12 @@
      * when all data is persisted to disk.
      * <p>
      * WAL state can be changed only for persistent caches.
+     * <p>
+     * <b>NOTE:</b>
+     * Currently, this method should only be called on a stable topology when no nodes are leaving or joining cluster,
+     * and all baseline nodes are present.
+     * Cache may be stuck in inconsistent state due to violation of these conditions. It is advised to destroy
+     * such cache.
```
Comment (Mmuzaf) [inline modules/core/src/main/java/org/apache/ignite/IgniteCluster.java:559]:
```suggestion
     * such cache.
      TODO: IGNITE-13976 WAL disable/enable with node restarts results in mismatching state, data loss
```

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/WalStateManager.java [hunk 4]
```diff
@@ -304,6 +305,12 @@ public boolean prohibitWALDisabling() {
             String.format("Cache WAL mode cannot be changed within lock or transaction " +
                     "[cacheNames=%s, walEnabled=%s]", cacheNames, enabled));
```
Comment (Mmuzaf) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/WalStateManager.java:307]:
```suggestion
     // TODO: IGNITE-13976 WAL disable/enable with node restarts results in mismatching state, data loss
```

**Review by Mmuzaf — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/ignite/IgniteCluster.java (hunk 1 above)
Comment (alamar) [reply modules/core/src/main/java/org/apache/ignite/IgniteCluster.java:532]: I don't think this is appropriate place for a TODO. We have this link in developer warning, where it is more convenient in my opinion.

TODOs usually reside in comments which are not javadoc.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/WalStateManager.java (hunk 4 above)
Comment (alamar) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/WalStateManager.java:307]: There's a link in the text below. But I can also add this comment. What do you think?

Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/WalStateManager.java:307]: Ilya, from my point of view we should add such comment due to we can forget to remove warnings when the issue will be fixed.

Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/WalStateManager.java:307]: But as you wish. Let's proceed anyway.

Comment (alamar) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/WalStateManager.java:307]: I think somebody will point out that the warning is still shown even through the issue is resolved. I will proceed.

**Pull request closed without merging**